backend/translations.js

```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+import fs from 'fs'
+import path from 'path'
+import util from 'util'
+
+import sbp from '~/shared/sbp.js'
+
+const readFileAsync = util.promisify(fs.readFile)
+
+const defaultLanguage = 'en-US'
+const languageDir = path.resolve('./strings')
+const languageFileExtension = '.json'
+const languageFileMap = new Map([
+  ['en', 'english.json'],
+  ['fr', 'french.json']
+])
+
+export default sbp('sbp/selectors/register', {
+  'backend/translations/get': async function (language: string): Promise<string> {
+    if (language === defaultLanguage) return ''
+    const [languageCode] = language.toLowerCase().split('-')
+    const fileName = languageFileMap.get(languageCode) || ''
+
+    // If we don't have a matching file yet,
+    // returns empty so that the UI stays in the default language.
+    if (!fileName || !fileName.endsWith(languageFileExtension)) return ''
```

Comment:
1. `languageFileExtension` is unnecessary because if the mapping exists it is guaranteed to end in `.json`.
2. If there is an error, then return an error, e.g. `Boom.notFound()`, not an empty string

Reply:
1. I agree that it's not very useful, in fact I've only used that constant to avoid embedding a string in the function body. Similarly, I've also defined `defaultLanguage` for the same reason, while the default language will likely never change from `en-US`.

2. I supposed it was better to return empty because I didn't consider that case to be an error or an exception, but rather to be the default. Quite in the same way that accessing a value from an object or map doesn't throw but returns a default value like `undefined` if the key was not found.
On the contrary, if a file name was found in `languageFileMap` but no real file was found or no data could be read, then an error certainly occured.

Reply:
Ah, I missed that this comment wasn't addressed in the followup commit. Feel free to tweek/optimize this in a future PR if you'd like!

Also, I'm about to merge updated strings after running `strings.mac`, see PR #964... although there are some issues I think @SGrondin needs to have a look at .... keep an eye on that PR